client/tx/factory.go

```diff
@@ -29,6 +30,7 @@ type Factory struct {
 	chainID            string
 	memo               string
 	fees               sdk.Coins
+	tip                *tx.Tip
```

Comment:
as proposed in https://github.com/cosmos/cosmos-sdk/issues/10264#issuecomment-979051872, let's think about not adding tips (and aux-related stuff) in the Factory.

What I'm thinking is:
- in `GenerateOrBroadcastTxCLI`, do a check:
    - if `--aux` flag is set, then call a private function `auxCLIHelper`
    - or else, create factory like before.